### PR TITLE
[gitlab] Migrate kitchen tests to k8s runners

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Migrates kitchen tests to the k8s Gitlab runners infrastructure.

Removes an old `rsync` line that's now useless (and made jobs fail in k8s runners).

### Motivation

Remove some load from the old `runner:main` infrastructure: most of the `datadog-agent` pipeline was running on `main` runners, which sometimes causes congestion during peak hours. By moving kitchen tests to k8s runners, we remove 6 jobs from each branch pipeline, and 70+ jobs from each `main` pipeline.

Migrate to a newer, more supported infrastructure.

### Describe how to test/QA your changes

Run a full pipeline, check that all kitchen tests work as expected: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/7428942.
*Note:* some security-agent kitchen tests failed, but these failures aren't due to the k8s runners migration.

Once this is merged: check that k8s jobs run well with the activity caused by the `datadog-agent` pipelines.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/functional_test/security_agent.yml [hunk 1]
```diff
@@ -30,7 +30,6 @@ kitchen_test_security_agent_x64:
   variables:
     KITCHEN_ARCH: x86_64
   before_script:
-    - rsync -azr --delete ./ $SRC_PATH
```
Comment (paulcacheux) [inline .gitlab/functional_test/security_agent.yml:33]: Why was this here, and why was this removed ?

**Review by paulcacheux — APPROVED** [review]
This is really great, thank you very much for working on this !

**Review by cvbarros — APPROVED** [review]

File: .gitlab/functional_test/security_agent.yml (hunk 1 above)
Comment (paulcacheux) [reply .gitlab/functional_test/security_agent.yml:33]: Just saw the message in the PR description, please ignore

**Review by albertvaka — APPROVED** [review]
Awesome 🚀 🚀 🚀

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -79,6 +79,8 @@ stages:
   - notify
 
 variables:
+  #Do not change this - must be the repository name for Kubernetes runners to work
+  KUBERNETES_SERVICE_ACCOUNT_OVERWRITE: "datadog-agent"
```
[the remark this answers is not in the document]
Comment (KSerrania) [reply .gitlab-ci.yml:83]: Good question for @cvbarros, I don't know the internals of the k8s runners enough to be able to answer this.

**Review by brycekahle — APPROVED** [review]
